### ISSUE TITLE
Coconut return code does not represent mypy reported errors

### ISSUE THREAD

**Issue by acdimalev** [evhub/coconut#388]
Even when mypy returns an error, Coconut's return code is zero.

```
$ coconut mypy-error.coco --mypy
Compiling         mypy-error.coco ...
Compiled to       mypy-error.py .
mypy-error.py:24: error: Argument 1 to "foo" has incompatible type "unicode"; expected "int"
$ echo $?
0
$ cat mypy-error.coco
def foo(a: int) -> str = str(int)

print(foo("one"))
```

**Comment by evhub** [comment]
@acdimalev This is now live on `coconut-develop`--just `pip install -U coconut-develop` to test it out.

**Comment by acdimalev** [comment]
Cool, this appears to be resolved by ed499ebe, with additional testing clean-up in ea5177bf.  Manual validation demonstrates a non-zero exit code, as expected.  Look forward to seeing this in the next release, thank you!

```
$ make
coconut mypy-error.coco -s --mypy
Compiling         mypy-error.coco ...
Compiled to       mypy-error.py .
mypy-error.py:24: error: Argument 1 to "foo" has incompatible type "unicode"; expected "int"
Exiting due to MyPy error.
make: *** [Makefile:7: mypy-error.py] Error 1
$ cat mypy-error.coco
def foo(a: int) -> str = str(int)

print(foo("one"))
```
